clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/consumer/AsyncHandlingAutoCommitKafkaConsumer.java

```diff
@@ -464,11 +514,12 @@ private synchronized void setCommittedOffsets(final Map<TopicPartition, OffsetAn
      */
     class TopicPartitionOffsets {
 
+        private static final long UNDEFINED_OFFSET = -2;
         private final TopicPartition topicPartition;
         private final Deque<OffsetsQueueEntry> queue = new LinkedList<>();
 
-        private long lastSequentiallyCompletedOffset = -1;
-        private long lastCommittedOffset = -1;
```

Comment:
Why are the defaults changed from `-1` to `-2`?

Reply:
The `TopicPartitionOffsets` inner class works with offsets that refer to the position of a received/handled record.
The offset for the actual offset commit is `+1` of that value.
With this PR, we now have the situation that we explicitly want to do an offset commit with position 0. That means that `lastSequentiallyCompletedOffset` has to be set to `-1` in that case, meaning `-1` can't be used as the number referring to an unknown offset anymore.